instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/HandlerThreadTest.kt

```diff
@@ -0,0 +1,88 @@
+package com.mapbox.navigation.instrumentation_tests.core
+
+import android.location.Location
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Looper
+import android.util.Log
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+import com.mapbox.navigation.core.internal.extensions.flowOnFinalDestinationArrival
+import com.mapbox.navigation.instrumentation_tests.activity.EmptyTestActivity
+import com.mapbox.navigation.instrumentation_tests.utils.MapboxNavigationRule
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.sdkTest
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.setNavigationRoutesAsync
+import com.mapbox.navigation.instrumentation_tests.utils.location.MockLocationReplayerRule
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider.toNavigationRoutes
+import com.mapbox.navigation.testing.ui.BaseTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.android.asCoroutineDispatcher
+import kotlinx.coroutines.flow.first
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class HandlerThreadTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java) {
+
+    override fun setupMockLocation(): Location = mockLocationUpdatesRule.generateLocationUpdate {
+        latitude = 38.894721
+        longitude = -77.031991
+    }
+
+    @get:Rule
+    val mockLocationReplayerRule = MockLocationReplayerRule(mockLocationUpdatesRule)
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var sdkThread: HandlerThread
+    private lateinit var handler: Handler
+    private lateinit var sdkDispatcher: CoroutineDispatcher
+
+    @Before
+    fun setUp() {
+        sdkThread = HandlerThread("test sdk thread")
+        sdkThread.start()
+        handler = Handler(sdkThread.looper)
+        sdkDispatcher = handler.asCoroutineDispatcher()
+        handler.post {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .build()
+            )
+        }
+    }
+
+    @After
+    fun tearDown() {
+        handler.post {
+            mapboxNavigation.onDestroy()
+        }
+        sdkThread.quitSafely()
```

Comment:
Why not `interrupt` or `quit`? The first one will abort all the `wait`, `join`, etc. The second one will not process any messages that are in the queue. While `quitSafely` will process all the messages and just will throw an exception when it receives new ones.
It's not very clear what you are trying to achieve here but I've always used `interrupt` to stop the threads. It's the most "forced" option.

Reply:
I just want to stop the SDK thread after the tests. Which practical benefits do you see in using `interrupt` over the `quiteSafely` in this specific case?

Reply:
Not in this specific case but in general I think we'd want to force-stop it. Here I don't think it really matters it's just that we shouldn't copy-paste this approach to production code without further exploring the options.